### PR TITLE
adapt RocksDB v9.1.1

### PR DESCRIPTION
**Pull request by yihuang**
~~there's a zstd cmake issue, fixed in https://github.com/facebook/rocksdb/pull/12309, either wait for the new release, or patch librocksdb locally.~~

### CODE REVIEW

**Comment by aaronyuan** [comment]
Is there any plan for this PR?

**Comment by linxGnu** [comment]
It's better to make changes for 9.1.x. I will give it a try

**Comment by aaronyuan** [comment]
> It's better to make changes for 9.1.x instead. I will give it a try

@linxGnu can we have a "adapt RocksDB v9.0.0" release? i am using the rocksdb 9.0.0 now. not easy to upgrade since there are customization of rocksdb.

**Comment by linxGnu** [comment]
@aaronyuan I am on it. 50% of code will be based on this PR

File: build.sh [hunk 1]
```diff
@@ -37,7 +37,7 @@ cd $BUILD_PATH && wget https://github.com/facebook/zstd/archive/v${zstd_version}
 
 # Note: if you don't have a good reason, please do not set -DPORTABLE=ON
 # This one is set here on purpose of compatibility with github action runtime processor
-rocksdb_version="8.11.3"
+rocksdb_version="9.0.0"
```
Comment (yihuang) [inline build.sh:40]:
```suggestion
rocksdb_version="9.1.1"
```

**Comment by yihuang** [comment]
close in favor of #151

**Comment by linxGnu** [comment]
@aaronyuan we released new version of grocksdb which is compatible with rocksdb 9.1.1. PTAL

**Comment by aaronyuan** [comment]
@linxGnu 

> @aaronyuan I am on it. 50% of code will be based on this PR

how about the other 50%, is it compatible with rocksdb 9.0.0. I suppose no. right?

**Pull request closed without merging**